transport/conn.go

```diff
@@ -214,6 +235,10 @@ type ConnConfig struct {
 const (
 	requestChanSize = 1024
 	ioBufferSize    = 8192
+
+	// Each handler may encounter 2 responses, one from connWriter.loop()
+	// and one from drainHandlers().
+	responseHandlerCapacity = 2
```

Comment:
I'd suggest to inline that and move the comment.
The comment here is lacking context and the value is not a subject to configuration.